### ISSUE TITLE
GcStress: Linux: Sanity Check failed in _il_dbgformat

### ISSUE THREAD

**Issue by swaroop-sridhar** [dotnet/coreclr#2764]
```
corerun JIT/Methodical/refany/_il_dbgformat/_il_dbgformat.exe

Assert failure(PID 19242 [0x00004b2a], Thread: 19242 [0x4b2a]): SanityCheck()
    File: /home/swaroops/git/a/coreclr/src/vm/methodtable.cpp Line: 9739
    Image: /home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

**** MessageBox invoked, title 'corerun - Assert Failure (PID 19242, Thread 19242/0x4b2a)' ****
  SanityCheck()

/home/swaroops/git/a/coreclr/src/vm/methodtable.cpp, Line: 9739

Abort - Kill program
Retry - Debug
Ignore - Keep running


Image:
/home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

********
```
Repro steps are:

* cp \scratch2\scratch\swaroops\coredistools\libcoredistools.so somewhere
* export LD_LIBRARY_PATH=somewhere
* export COMPlus_GCStress=0xC
* Run the above test

Here are the list of mkrefany tests that are failing for the same reason
JIT/Methodical/refany/_il_dbgformat/_il_dbgformat.exe
JIT/Methodical/refany/_il_dbglcs/_il_dbglcs.exe
JIT/Methodical/refany/_il_dbgshortsig/_il_dbgshortsig
JIT/Methodical/refany/_il_dbglongsig/_il_dbglongsig.exe
JIT/Methodical/refany/_il_rellcs/_il_rellcs.exe
JIT/Methodical/VT/etc/_relgc_nested/_relgc_nested.sh

**Comment by sivarv** [comment]
mkrefany test case. Fix to correctly handle TypedReference will fix this.

**Comment by LLITCHEV** [comment]
Fixed by PR #2860
